### PR TITLE
Implement tuple types

### PR DESCRIPTION
**Pull request by dfaust**
This adds a new `Tuple` type, which is very similar to the `UDT` type. It has to be used in conjunction with a rust `struct` and the user has to implement `Into<Bytes>` and `try_from(tuple: Tuple)`. It would be impossible to implement all conversions in `cdrs` because the size of a tuple is pretty much unlimited. Even implementing it for only a few items would be extreme.
Have a look at the tests to see it in action.

There are some parts of the code I didn't understand and just copied from the `UDT` type like in `fn decode_tuple` and `impl FromCursor for CTuple`.

This PR relies on #154 and don't worry, I can rebase this PR after you merged #154.

### CODE REVIEW

File: tests/common/mod.rs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+               AND durable_writes = false";
+    let query = QueryBuilder::new(cql).finalize();
+    session.query(query, true, true)?;
+
+    for create_cql in create_cqls.iter() {
+        let table_name = re_table_name
+            .captures(create_cql)
+            .map(|cap| cap.get(1).unwrap().as_str());
+
+        // if let Some(table_name) = table_name {
```
Comment (AlexPikalov) [inline tests/common/mod.rs]: Could you remove it or add any comments why we're keeping it.

File: tests/native_types.rs [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+    let query = QueryBuilder::new(cql).finalize();
+    let rows = session
+        .query(query, false, false)
+        .expect("query")
+        .get_body()
+        .expect("get body")
+        .into_rows()
+        .expect("into rows");
+    assert_eq!(rows.len(), 1);
+    for row in rows {
```
Comment (AlexPikalov) [inline tests/native_types.rs]: Please add an empty row upfront the `assert_eq!` as a separator between query execution and checking result sections

File: tests/native_types.rs [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+    let cql = "SELECT * FROM cdrs_test.test_integer_v4";
+    let query = QueryBuilder::new(cql).finalize();
+    let rows = session
+        .query(query, false, false)
+        .expect("query")
+        .get_body()
+        .expect("get body")
+        .into_rows()
+        .expect("into rows");
+    assert_eq!(rows.len(), 1);
```
Comment (AlexPikalov) [inline tests/native_types.rs]: the same

File: tests/native_types.rs [hunk 4]
```diff
@@ -0,0 +1,297 @@
...
+    let cql = "SELECT * FROM cdrs_test.test_float";
+    let query = QueryBuilder::new(cql).finalize();
+    let rows = session
+        .query(query, false, false)
+        .expect("query")
+        .get_body()
+        .expect("get body")
+        .into_rows()
+        .expect("into rows");
+    assert_eq!(rows.len(), 1);
```
Comment (AlexPikalov) [inline tests/native_types.rs]: as above

File: tests/native_types.rs [hunk 5]
```diff
@@ -0,0 +1,297 @@
...
+    let cql = "SELECT * FROM cdrs_test.test_blob";
+    let query = QueryBuilder::new(cql).finalize();
+    let rows = session
+        .query(query, false, false)
+        .expect("query")
+        .get_body()
+        .expect("get body")
+        .into_rows()
+        .expect("into rows");
+    assert_eq!(rows.len(), 1);
```
Comment (AlexPikalov) [inline tests/native_types.rs]: as above

File: tests/native_types.rs [hunk 6]
```diff
@@ -0,0 +1,297 @@
...
+    let cql = "SELECT * FROM cdrs_test.test_uuid";
+    let query = QueryBuilder::new(cql).finalize();
+    let rows = session
+        .query(query, false, false)
+        .expect("query")
+        .get_body()
+        .expect("get body")
+        .into_rows()
+        .expect("into rows");
+    assert_eq!(rows.len(), 1);
```
Comment (AlexPikalov) [inline tests/native_types.rs]: as above

File: tests/native_types.rs [hunk 7]
```diff
@@ -0,0 +1,297 @@
...
+    let cql = "SELECT * FROM cdrs_test.test_time";
+    let query = QueryBuilder::new(cql).finalize();
+    let rows = session
+        .query(query, false, false)
+        .expect("query")
+        .get_body()
+        .expect("get body")
+        .into_rows()
+        .expect("into rows");
+    assert_eq!(rows.len(), 1);
```
Comment (AlexPikalov) [inline tests/native_types.rs]: the same

File: tests/native_types.rs [hunk 8]
```diff
@@ -0,0 +1,297 @@
...
+    let cql = "SELECT * FROM cdrs_test.test_inet";
+    let query = QueryBuilder::new(cql).finalize();
+    let rows = session
+        .query(query, false, false)
+        .expect("query")
+        .get_body()
+        .expect("get body")
+        .into_rows()
+        .expect("into rows");
+    assert_eq!(rows.len(), 1);
```
Comment (AlexPikalov) [inline tests/native_types.rs]: as above

File: tests/tuple_types.rs [hunk 9]
```diff
@@ -0,0 +1,179 @@
...
+    let cql = "SELECT * FROM cdrs_test.simple_tuple";
+    let query = QueryBuilder::new(cql).finalize();
+    let rows = session
+        .query(query, false, false)
+        .expect("query")
+        .get_body()
+        .expect("get body")
+        .into_rows()
+        .expect("into rows");
+    assert_eq!(rows.len(), 1);
```
Comment (AlexPikalov) [inline tests/tuple_types.rs]: as above

File: tests/tuple_types.rs [hunk 10]
```diff
@@ -0,0 +1,179 @@
...
+    let cql = "SELECT * FROM cdrs_test.test_nested_tuples";
+    let query = QueryBuilder::new(cql).finalize();
+    let rows = session
+        .query(query, false, false)
+        .expect("query")
+        .get_body()
+        .expect("get body")
+        .into_rows()
+        .expect("into rows");
+    assert_eq!(rows.len(), 1);
```
Comment (AlexPikalov) [inline tests/tuple_types.rs]: as above

File: tests/user_defined_types.rs [hunk 11]
```diff
@@ -0,0 +1,147 @@
...
+    let cql = "SELECT * FROM cdrs_test.test_simple_udt";
+    let query = QueryBuilder::new(cql).finalize();
+    let rows = session
+        .query(query, false, false)
+        .expect("query")
+        .get_body()
+        .expect("get body")
+        .into_rows()
+        .expect("into rows");
+    assert_eq!(rows.len(), 1);
```
Comment (AlexPikalov) [inline tests/user_defined_types.rs]: the same

File: tests/user_defined_types.rs [hunk 12]
```diff
@@ -0,0 +1,147 @@
...
+    let cql = "SELECT * FROM cdrs_test.test_nested_udt";
+    let query = QueryBuilder::new(cql).finalize();
+    let rows = session
+        .query(query, false, false)
+        .expect("query")
+        .get_body()
+        .expect("get body")
+        .into_rows()
+        .expect("into rows");
+    assert_eq!(rows.len(), 1);
```
Comment (AlexPikalov) [inline tests/user_defined_types.rs]: the same as above

**Comment by dfaust** [comment]
Alright, tests are green now.

**Comment by AlexPikalov** [comment]
Great! Merged.

Thank you

**Pull request merged**